src/link/Coff.zig

```diff
@@ -3602,104 +3617,53 @@ fn freeRelocations(coff: *Coff, atom_index: Atom.Index) void {
-        itab.free_list.deinit(allocator);
-        itab.lookup.deinit(allocator);
+    pub fn getLibName(import: Import, coff: *const Coff) []const u8 {
+        return coff.temp_strtab.getAssumeExists(import.lib_name);
     }
 
-    /// Size of the import table does not include the sentinel.
-    fn size(itab: ImportTable) u32 {
-        return @as(u32, @intCast(itab.entries.items.len)) * @sizeOf(u64);
+    pub fn getSymbolName(import: Import, coff: *const Coff) []const u8 {
+        return coff.getSymbolName(import.symbol);
     }
 
-    fn addImport(itab: *ImportTable, allocator: Allocator, target: SymbolWithLoc) !ImportIndex {
-        try itab.entries.ensureUnusedCapacity(allocator, 1);
-        const index: u32 = blk: {
-            if (itab.free_list.popOrNull()) |index| {
-                log.debug("  (reusing import entry index {d})", .{index});
-                break :blk index;
-            } else {
-                log.debug("  (allocating import entry at index {d})", .{itab.entries.items.len});
-                const index = @as(u32, @intCast(itab.entries.items.len));
-                _ = itab.entries.addOneAssumeCapacity();
-                break :blk index;
-            }
-        };
-        itab.entries.items[index] = target;
-        try itab.lookup.putNoClobber(allocator, target, index);
-        return index;
+    pub fn getSymbol(import: Import, coff: *const Coff) *const coff_util.Symbol {
+        return coff.getSymbol(import.symbol);
     }
 
-    const Context = struct {
-        coff: *const Coff,
-        /// Index of this ImportTable in a global list of all tables.
-        /// This is required in order to calculate the base vaddr of this ImportTable.
-        index: usize,
-        /// Offset into the string interning table of the DLL this ImportTable corresponds to.
-        name_off: u32,
-    };
-
-    fn getBaseAddress(ctx: Context) u32 {
-        const header = ctx.coff.sections.items(.header)[ctx.coff.idata_section_index.?];
-        var addr = header.virtual_address;
-        for (ctx.coff.import_tables.values(), 0..) |other_itab, i| {
-            if (ctx.index == i) break;
-            addr += @as(u32, @intCast(other_itab.entries.items.len * @sizeOf(u64))) + 8;
-        }
-        return addr;
+    pub fn getSymbolPtr(import: Import, coff: *Coff) *coff_util.Symbol {
+        return coff.getSymbolPtr(import.symbol);
     }
 
-    fn getImportAddress(itab: *const ImportTable, target: SymbolWithLoc, ctx: Context) ?u32 {
-        const index = itab.lookup.get(target) orelse return null;
-        const base_vaddr = getBaseAddress(ctx);
-        return base_vaddr + index * @sizeOf(u64);
+    pub fn getAddress(import: Import, coff: *const Coff) u32 {
+        const header = coff.sections.items(.header)[coff.idata_section_index.?];
+        return header.virtual_address + import.rva;
     }
 
-    const FormatContext = struct {
-        itab: ImportTable,
-        ctx: Context,
-    };
+    const Index = u32;
```

Comment:
Please make this a non-exhaustive enum for type safety.

Reply:
If I understand your request correctly, this would imply `const Index = enum(u32) { _ };`. Can you explain how this is more type safe than `const Index = u32;` when we do not have any special indexes yet?

Reply:
In short, it prevents you from confusing `u32`s with different meanings. Making this type a non-exhaustive enum means:
* You cannot accidentally use values of this type in place of some other 32-bit value, e.g. a length
* You cannot accidentally use some other 32-bit value in place of an `Index`

As a random example: `InternPool.Cau.Index` and `InternPool.Nav.Index` are both non-exhaustive enums with no special fields, but the existence of those types means I will get a compile error if I try to use an `InternPool.Cau.Index` (or a `u32`, or an `InternPool.Index`, etc)  where I expected an `InternPool.Nav.Index`.

Another way to think about this is to consider that indices are quite similar to pointers, so encoding information about what an index "points to" in the type system is safer in just the same way as it is safer to have types `*T` rather than collapsing them all under one type `pointer`.

Reply:
Makes sense, thanks!

Reply:
Another place that could use this safety is std.zig.Ast.Node.Index - it is a u32, and intermixed with "other u32s" - what's special about this one is that the value 0 corresponds to "null_node", and is checked all over the place. Definitely a bag of worms. Not relevant to this PR, but thought I would mention it.